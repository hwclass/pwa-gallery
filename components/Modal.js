export default class extends React.Component {
  onClose(e) {
    this.props.onClose()
  }

  render() {
    const { id, uri, title } = this.props
    return (
      <div className='frame' onClick={(e) => this.onClose(e)}>
        <span className="title">
          {title}
          <a href="#" className="closeIcon"/>
        </span>
        <picture key={id} className="pic">
          <img src={`http://${uri}_27.jpg`} />
        </picture>
        <style jsx>{`
          .frame {
            position: fixed;
            background: rgba(0,0,0,1);
            left: 0;
            right: 0;
            top: 0;
            bottom: 0;
            margin: auto;
          }
          .title {
            display: block;
            max-width: 100%;
            padding: 15px;
            text-align: center;
            font-size: 2.5vmin;
            background: #000;
            color: #FFF;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
          }
          @media screen and (max-width: 400px) {
            .title {
              font-size: 2.5vmax;
            }
          }
          .closeIcon {
            position: absolute;
            right: 20px;
            top: 8px;
            width: 22px;
            height: 22px;
            opacity: 1;
          }
          @media screen and (max-width: 400px) {
            .closeIcon {
              max-width: 100%;
              top: 9px;
              width: 21px;
              height: 5px;
            }
          }
          .closeIcon:before, .closeIcon:after {
            position: absolute;
            left: 15px;
            content: ' ';
            height: 33px;
            width: 2px;
            background-color: #333;
          }
          .closeIcon:before {
            transform: rotate(45deg);
          }
          .closeIcon:after {
            transform: rotate(-45deg);
          }
          .pic {
            position: absolute;
            margin: 0 auto;
            display: block;
            left: 50%;
            top: 50%;
            transform: translate(-50%, -50%);
            width: 50%;
            padding: 20px; 
          }
          @media screen and (max-width: 400px) {
            .pic {
              padding: 0;
              width: 100%;
            }
          }
          img {
            display: block;
            margin: auto;
            width: 100%;
          }
          @media screen and (max-width: 400px) {
            img {
              width: 100%;
            }
          }
        `}</style>
      </div>
    );
  }
}