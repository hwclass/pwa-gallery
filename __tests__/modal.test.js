/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Modal from '../components/Modal'

describe('Modal', () => {
  it('should render', () => {
    const component = shallow(<Modal id='1' uri='/some-url/for-image-1.jpg' title='Modal Title'/>)
    expect(component.getElements()).toMatchSnapshot();
  })
})

describe('Modal', () => {
  it('should match with snapshot', () => {
    const component = renderer.create(<Modal id='1' uri='/some-url/for-image-1.jpg' title='Modal Title'/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})