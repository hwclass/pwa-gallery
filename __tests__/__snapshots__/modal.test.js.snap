// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match with snapshot 1`] = `
<div
  className="jsx-475590090 frame"
  onClick={[Function]}
  role="presentation"
>
  <span
    className="jsx-475590090 title"
  >
    Modal Title
    <button
      className="jsx-475590090 closeIcon"
    />
  </span>
  <picture
    className="jsx-475590090 pic"
  >
    <img
      className="jsx-475590090"
      src="http:///some-url/for-image-1.jpg_27.jpg"
    />
  </picture>
</div>
`;

exports[`Modal should render 1`] = `
Array [
  <div
    className="jsx-475590090 frame"
    onClick={[Function]}
    role="presentation"
  >
    <span
      className="jsx-475590090 title"
    >
      Modal Title
      <button
        className="jsx-475590090 closeIcon"
      />
    </span>
    <picture
      className="jsx-475590090 pic"
    >
      <img
        className="jsx-475590090"
        src="http:///some-url/for-image-1.jpg_27.jpg"
      />
    </picture>
    <JSXStyle
      css=".frame.jsx-475590090{position:fixed;background:rgba(0,0,0,1);left:0;right:0;top:0;bottom:0;margin:auto;}.title.jsx-475590090{display:block;max-width:100%;padding:15px;text-align:center;font-size:2.5vmin;background:#000;color:#FFF;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;}@media screen and (max-width:400px){.title.jsx-475590090{font-size:2.5vmax;}}.closeIcon.jsx-475590090{position:absolute;right:20px;top:8px;width:22px;height:22px;opacity:1;background:transparent;border:none;cursor:pointer;}@media screen and (max-width:400px){.closeIcon.jsx-475590090{max-width:100%;top:9px;width:21px;height:5px;}}.closeIcon.jsx-475590090:before,.closeIcon.jsx-475590090:after{position:absolute;left:15px;content:' ';height:33px;width:2px;background-color:#333;}.closeIcon.jsx-475590090:before{-webkit-transform:rotate(45deg);-ms-transform:rotate(45deg);transform:rotate(45deg);}.closeIcon.jsx-475590090:after{-webkit-transform:rotate(-45deg);-ms-transform:rotate(-45deg);transform:rotate(-45deg);}.pic.jsx-475590090{position:absolute;margin:0 auto;display:block;left:50%;top:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);width:50%;padding:20px;}@media screen and (max-width:400px){.pic.jsx-475590090{padding:0;width:100%;}}img.jsx-475590090{display:block;margin:auto;width:100%;}@media screen and (max-width:400px){img.jsx-475590090{width:100%;}}"
      styleId="475590090"
    />
  </div>,
]
`;
